### ISSUE TITLE
vue3 importmap workaround does not appear to work with vue.global.prod.js/vue.global.min.js

### ISSUE THREAD

**Issue by adq** [systemjs/systemjs#2300]
**TL;DR**
Production/minified vue3 builds fail for me:
- https://cdn.jsdelivr.net/npm/vue@3.0.5/dist/vue.global.js --> OK, but its non-minified non-prod code
- https://cdn.jsdelivr.net/npm/vue@3.0.5/dist/vue.global.min.js --> FAILS, its the auto-minified(?) version of the above, but its still non-prod code (it prints a message saying NOT PRODUCTION even tho its minified).
- https://cdn.jsdelivr.net/npm/vue@3.0.5/dist/vue.global.prod.js --> FAILS differently, its the proper vue production release code

**Detail**
Hi, I set up the fix for vue3 as an importmapped script here (many thanks BTW): https://github.com/systemjs/systemjs/issues/2272#issuecomment-744636282

I updated it to vue 3.0.5 and all is well:
```
<script type="systemjs-importmap">
    {
      "imports": {
        "vue": "https://cdn.jsdelivr.net/npm/vue@3.0.5/dist/vue.global.js",
      }
    }
</script>
<script src="https://cdn.jsdelivr.net/npm/vue@3.0.5/dist/vue.global.js"></script>
<script>
    Object.getPrototypeOf(System).firstGlobalProp = true;
    System.set(System.resolve('vue'), window.Vue)
    System.import('@tvsquared/tvs-spapoc-rootconfig');
</script>
```

BUT when I switch to production vue code (vue.global.prod.js) it fails with another error:
```
app-errors.js:12 Uncaught ReferenceError: application '@tvsquared/tvs-spapoc-nav' died in status LOADING_SOURCE_CODE: __VUE_HMR_RUNTIME__ is not defined
    at eval (App.vue?dfb6:8)
    at Module../src/App.vue (app.js:2348)
    at __webpack_require__ (app.js:800)
    at fn (app.js:111)
    at eval (main.js:16)
    at Module../src/main.js (app.js:2384)
    at __webpack_require__ (app.js:800)
    at fn (app.js:111)
    at Object.1 (app.js:2398)
    at __webpack_require__ (app.js:800)
```

I noted the example vue3 project, https://github.com/musashiM82/root-config/blob/main/src/index.ejs
uses `https://cdn.jsdelivr.net/npm/vue@3.0.3/dist/vue.global.min.js`.  (I'm not sure what the difference is between prod.js and min.js)

So I tried using https://cdn.jsdelivr.net/npm/vue@3.0.5/dist/vue.global.min.js, even though vue.global.min.js does not show up in the file listing at https://cdn.jsdelivr.net/npm/vue@3.0.5/dist/ (maybe they have auto-minification?)

I get a different error:
```
app-errors.js:12 Uncaught TypeError: application '@tvsquared/tvs-spapoc-nav' died in status LOADING_SOURCE_CODE: Cannot read property 'createVNode' of undefined
    at eval (App.vue?3dfd:3)
    at Module../node_modules/cache-loader/dist/cjs.js?!./node_modules/babel-loader/lib/index.js!./node_modules/vue-loader-v16/dist/templateLoader.js?!./node_modules/cache-loader/dist/cjs.js?!./node_modules/vue-loader-v16/dist/index.js?!./src/App.vue?vue&type=template&id=7ba5bd90 (app.js:904)
    at __webpack_require__ (app.js:800)
    at fn (app.js:111)
    at eval (App.vue?4907:1)
    at Module../src/App.vue?vue&type=template&id=7ba5bd90 (app.js:2372)
    at __webpack_require__ (app.js:800)
    at fn (app.js:111)
    at eval (App.vue?dfb6:1)
    at Module../src/App.vue (app.js:2348)
```

Just in case I'm going mad, I went back to https://cdn.jsdelivr.net/npm/vue@3.0.5/dist/vue.global.js -- it still works ok.

**Comment by joeldenning** [comment]
Try deleting this line, as it's not needed when using the System.set approach:

>     Object.getPrototypeOf(System).firstGlobalProp = true;

**Comment by joeldenning** [comment]
> _VUE_HMR_RUNTIME__ is not defined

Vue hot module reloading (HMR) is not included in the production version of vue 3. In your case, it seems you're likely developing a Vue 3 app in development mode with HMR enabled, on an html file that has the production Vue 3 code.

**Comment by guybedford** [comment]
Closing due to inactivity.
